postgresql/resource_postgresql_schema.go

```diff
@@ -110,8 +117,24 @@ func resourcePostgreSQLSchemaCreate(d *schema.ResourceData, meta interface{}) er
 
 	queries := []string{}
 
+	database := getDatabase(d, c)
+
+	c.catalogLock.Lock()
+	defer c.catalogLock.Unlock()
+
+	txn, err := startTransaction(c, database)
+	if err != nil {
+		return err
+	}
+	defer deferredRollback(txn)
+
 	schemaName := d.Get(schemaNameAttr).(string)
-	{
+
+	// Check if previous tasks haven't already create schema
```

Comment:
What do you mean by previous tasks?

I need to think about that.... Even if I can understand why it could be useful, normally Terraform providers usually don't check if resources exist  before creating.
We usually use `terraform import` if we create a definition of an existing resource.
(Otherwise may not even realize that the resource he created actually  already exists....)

Reply:
yes, the `public` schema is a good reason to do this as it's created with the database :+1: